facia-tool/public/js/models/config/persistence.js

```diff
@@ -1,114 +1,137 @@
 
-    function serializeFront (front) {
-        var model = flattenModel(front.props);
+        return accumulator;
+    }, {});
+}
 
-        model.id = front.id();
+function serializeCollection (collection) {
+    var model = flattenModel(collection.meta);
 
-        model.collections = _.chain(front.collections.items()).map(function (collection) {
-            return collection.id;
-        }).filter(identity).value();
-        return model;
+    if (collection.id) {
+        model.id = collection.id;
     }
 
-    return {
-        collection: {
-            create: function (collection) {
-                postUpdate({
-                    url: '/config/collections',
-                    data: JSON.stringify({
-                        frontIds: collection.frontIds(),
-                        collection: serializeCollection(collection)
-                    })
-                });
-            },
-
-            update: function (collection) {
-                postUpdate({
-                    url: '/config/collections/' + collection.id,
-                    data: JSON.stringify({
-                        frontIds: collection.frontIds(),
-                        collection: serializeCollection(collection)
-                    })
-                });
-            }
-        },
+    return model;
+}
 
-        front: {
-            create: function (front, collection) {
-                var serialized = serializeFront(front);
+function serializeFront (front) {
+    var model = flattenModel(front.props);
 
-                /** We instead pass through the initial collection as a separate parameter */
-                delete serialized.collections;
+    model.id = front.id();
 
-                postUpdate({
-                    url: '/config/fronts',
-                    data: JSON.stringify(_.extend({
-                        initialCollection: serializeCollection(collection)
-                    }, serialized))
-                });
-            },
+    model.collections = _.chain(front.collections.items()).map(function (collection) {
+        return collection.id;
+    }).filter(identity).value();
+    return model;
+}
 
-            update: function (front) {
-                postUpdate({
-                    url: '/config/fronts/' + front.id(),
-                    data: JSON.stringify(serializeFront(front))
-                });
-            }
-        },
+function isInitialCollection (collection) {
+    var parents = collection.parents();
 
-        registerCallback: function (onUpdate) {
-            onUpdateCallbacks.push(onUpdate);
-        }
-    };
-});
+    if (parents.length === 1) {
+        var siblings = parents[0].collections.items();
+
+        return siblings.length === 1 && siblings[0] === collection;
+    } else {
+        return false;
+    }
+}
+
+function postUpdate (opts) {
+    return publicInterface[postUpdateSym](opts);
+}
+
+function createFrontFromCollection (collection) {
+    var front = collection.parents()[0],
```

Comment:
i think `const` is misleading and useless. It's a `const` reference, not a `const` value, people reading `const front` might expect that `front` doesn't change, but it does.